torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -353,6 +353,47 @@ Tensor _nested_from_padded_backward(
   return grad.to_padded_tensor(0, input.sizes());
 }
 
+std::tuple<Tensor, Tensor, Tensor> NestedTensor_linear_backward(
+    const Tensor& grad,
+    const Tensor& self,
+    const Tensor& weight,
+    const c10::optional<Tensor>& bias_opt) {
+  if (!grad.defined()) {
+    return std::tuple<Tensor, Tensor, Tensor>{Tensor(), Tensor(), Tensor()};
+  }
+  auto* nt_grad = at::native::get_nested_tensor_impl_or_null(grad);
+  auto* nt_self = at::native::get_nested_tensor_impl_or_null(self);
+  TORCH_CHECK(nested_tensor_impl_is_contiguous(nt_grad));
+  auto grad_buffer = nt_grad->get_buffer();
+  auto self_buffer = nt_self->get_buffer();
+
+  auto d_weight = at::mm(
+      grad_buffer.reshape({-1, weight.size(0)}).t_(),
```

Comment:
save the reshaped version of grad_buffer since that's the only version of grad_buffer we need?